lib/lib.go

```diff
@@ -294,6 +296,22 @@ func OptProfiles(pros profile.Store) Option {
 	}
 }
 
+// OptBus supplies a bus for the instance
+func OptBus(bus event.Bus) Option {
+	return func(o *InstanceOptions) error {
+		o.bus = bus
+		return nil
+	}
+}
+
+// OptDscache supplies a Dscache for the instance
+func OptDscache(dscache *dscache.Dscache) Option {
+	return func(o *InstanceOptions) error {
+		o.dscache = dscache
+		return nil
+	}
+}
+
```

Comment:
It's possible these are too... low level for us to reasonably want someone to add their own, however they were extremely helpful in debugging this issue!

Reply:
I think it's fine to have these options, but they should have documenting comments that clarifies these options _override defaults_.